src/dns_parser.rs

```diff
@@ -7,7 +7,10 @@
 #[cfg(feature = "logging")]
 use crate::log::trace;
 
-use crate::error::{e_fmt, Error, Result};
+use crate::{
+    error::{e_fmt, Error, Result},
+    service_info::InterfaceScope,
```

Comment:
@keepsimple1: `dns_parser` is ought to be a separate module, do we want to provide `InterfaceId` from `dns_parser`. Or would it be better create a separate module `interface_id` which provides `InterfaceId` for `dns_parser` and `service_info`. I'm not that fluent yet in rust best practices about modules, I'm just guessing from experience from other languages and I sort of would prefer the `interface_id` module.

Reply:
I think it's okay to define `InterfaceId` in `dns_parser`.  We can refactor the module later once we have clearer understanding.